### PR TITLE
Feat/invite_registration

### PR DESCRIPTION
**Pull request by ajhollid**
Implemented registration from email with token.

Please double check my updates to the validation, I think I've got it straight but I'd apprecaite more eyes on it

### CODE REVIEW

File: Server/db/MongoDB.js [hunk 1]
```diff
@@ -185,6 +185,21 @@ const requestInviteToken = async (req, res) => {
   }
 };
 
+const getInviteToken = async (req, res) => {
+  try {
+    console.log(req.body.token);
+    const invite = await InviteToken.findOneAndDelete({
+      token: req.body.token,
+    });
+    if (invite === null) {
```
Comment (Skorpios604) [inline Server/db/MongoDB.js:193]: maybe 
```javascript 
if (!invite)
```

File: Server/controllers/authController.js [hunk 2]
```diff
@@ -237,6 +238,21 @@ const inviteController = async (req, res, next) => {
...
+  try {
+    await inviteVerifciationBodyValidation.validateAsync(req.body);
+    const invite = await req.db.getInviteToken(req, res);
+
+    res
+      .status(200)
+      .json({ status: "success", msg: "Invite verified", data: invite });
+  } catch (error) {
+    error.service = SERVICE_NAME;
+    next(error);
```
Comment (Skorpios604) [inline Server/controllers/authController.js:252]: Function will end at ```next(error);```
No need for a return here I don't think.

**Review by Skorpios604 — APPROVED** [review]

File: Server/db/MongoDB.js (hunk 1 above)
Comment (ajhollid) [reply Server/db/MongoDB.js:193]: I always prefer strict evaluation whenver possible.  

`!invite` allows for all sorts of things like `0`, `""`, 'undefined`, `NaN` etc that I would definitely _not_ want passing that check :P

Comment (Skorpios604) [reply Server/db/MongoDB.js:193]: Good point!

File: Server/controllers/authController.js (hunk 2 above)
Comment (ajhollid) [reply Server/controllers/authController.js:252]: I belive you are correct here, the return is left over from an earlier function I wrote that had a conditional `next()` call and I wanted to make sure no further code was executed after that so I added the `return` and now I've just gotten in the habit of doing it to be safe.

**Pull request merged**